rfc/helm-synthesis.md

```diff
@@ -0,0 +1,205 @@
+
+---
+
+## Internal FAQ
+
+### Why are we doing this?
+
+The motivation for this change is to ease the deployment process for the users by providing them with `helm` as a synthesis format option.
+
+### Why should we _not_ do this?
+
+There are no downside of adding this features. This would add on to the value of our product.
+
+### What is the technical solution (design) of this feature?
+
+CDK8s CLI currently provides users with a `synth` command that helps with generating manifests for the given CDK8s app. This design focuses on adding more functionality to this command, like this feature would enable users to generate manifests in a structure that is supported by helm and can ease the helm deployment with CDK8s experience for the user.
+
+For instance, once implemented the user would be able to run,
+```
+cdk8s synth --format helm --output ./chart && helm install <release-name> ./chart
+```
+where,
+* `cdk8s synth`: Is the synth command provided by CDK8s CLI for synthesizing a CDK8s application.
+* `--format`: **[NEW]** This would be the flag that can take helm as an option. This would mean that the manifests we would generate for the user would be structured in such a format that it's easier to deploy with helm. By default, this would be cdk8s, which means synthesis would take place as usual.
+* `--output`: This is an existing flag where user can specify in which directory they would like the generated manifests to be stored in.
+* `helm install`:
+   - [Command](https://helm.sh/docs/helm/helm_install/) to deploy manifests hosted within a helm chart to Kubernetes cluster. 
+   - `<release-name>`: The release name for the helm deployment.
+   - `./chart`: Is the folder here where we have the generated helm chart by CDK8s app.
+
+_An example is worth a thousand words_. Let's go through a couple of user scenarios to understand more about the design being proposed.
+
+**Synthesize CDK8s Apps into Helm Charts**
+
+**Assumptions**
+- The user is running synthesis with `--format helm` for a [CDK8s app and not a CDK8s library](##Appendix).
+
+**User runs `cdk8s synth --format helm --output ./chart`** 
+
+![](./images/helm-synthesis.png)
+
+For helm to consume our generated manifests for deployment, we would need to generate a structure similar to what [Helm Charts](https://v2.helm.sh/docs/developing_charts/) look like. The following is a simpler folder structure complying with helm structure. The user can add more to it if they desire but this would work for our purpose.
+
+```
+chart/              # Value of --output. Defaults to directoryName if format is helm
+├── Chart.yaml      # [REQUIRED] Information about your chart
+├── Readme.md       # A generic readme conveying this chart is generated by CDK8s
+└── templates/      # The template files. These would contain the generated manifest files
```

Comment:
Helm has a cool [feature for supporting CRDs](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/). I think it would be really awesome to support that. Note that the way they support it is different in helm v3 vs helm v2. 

cdk8s could read the `imports` section of the config file, find the CRDs, and just place them in the `crds` directory.